examples/aws/terraform/ha-autoscale-cluster/auth_asg.tf

```diff
@@ -33,6 +33,26 @@ resource "aws_autoscaling_group" "auth" {
     propagate_at_launch = true
   }
 
+  dynamic "tag" {
+    for_each = data.aws_default_tags.this.tags
+    content {
+      key                 = tag.key
+      value               = tag.value
+      propagate_at_launch = true
+    }
+  }
+
+  dynamic "instance_refresh" {
+    for_each = var.enable_auth_asg_instance_refresh ? [1] : []
+    content {
+      strategy = "Rolling"
+      preferences {
+        auto_rollback          = true
+        min_healthy_percentage = 50
+      }
+    }
+  }
```

Comment:
We don't technically support rolling restart for Auth. For proxy it's OK, but Auth should all be restarted at the same time during updates. I don't think there's an easy way to do it, maybe `min_healthy_percentage = 0`.

Also we really don't want to automatically rollback auth as not all updates can be rolled back. You might have to rollback the state at the same time.

Reply:
This is a good point, but we also tell customers to scale auth servers to 1 when they upgrade or downgrade Teleport right?

Is there any problem with a rolling instance refresh when configuration *other than teleport version* is changed?